service/worker/addsearchattributes/workflow.go

```diff
@@ -0,0 +1,203 @@
+	return nil
+}
+
+func (a *activities) AddESMappingFieldActivity(ctx context.Context, params WorkflowParams) error {
+	mapping := make(map[string]string, len(params.CustomAttributesToAdd))
+	for saName, saType := range params.CustomAttributesToAdd {
+		esType := searchattribute.MapESType(saType)
+		if esType == "" {
+			return temporal.NewNonRetryableApplicationError(fmt.Sprintf("Unknown search attribute type: %v", saType), "", nil)
+		}
+		mapping[saName] = esType
+	}
+
+	a.logger.Info("Creating Elasticsearch mapping.", tag.ESIndex(params.IndexName), tag.ESMapping(mapping))
+	_, err := a.esClient.PutMapping(ctx, params.IndexName, searchattribute.Attr, mapping)
+	if err != nil {
+		a.metricsClient.IncCounter(metrics.AddSearchAttributesWorkflowScope, metrics.AddSearchAttributesFailuresCount)
+		if esclient.IsRetryableError(err) {
+			a.logger.Error("Unable to update Elasticsearch mapping (retryable error).", tag.ESIndex(params.IndexName), tag.Error(err))
+			return fmt.Errorf("%w: %v", ErrUnableToUpdateESMapping, err)
+		}
+		a.logger.Error("Unable to update Elasticsearch mapping (non-retryable error).", tag.ESIndex(params.IndexName), tag.Error(err))
+		return temporal.NewNonRetryableApplicationError(fmt.Sprintf("%v: %v", ErrUnableToUpdateESMapping, err), "", nil)
+	}
+	a.logger.Info("Elasticsearch mapping created.", tag.ESIndex(params.IndexName), tag.ESMapping(mapping))
+
+	return nil
+}
+
+func (a *activities) WaitForYellowStatusActivity(ctx context.Context, index string) error {
+	status, err := a.esClient.WaitForYellowStatus(ctx, index)
+	if err != nil {
+		a.logger.Error("Unable to get Elasticsearch cluster status.", tag.ESIndex(index), tag.Error(err))
+		a.metricsClient.IncCounter(metrics.AddSearchAttributesWorkflowScope, metrics.AddSearchAttributesFailuresCount)
+		return err
+	}
+	a.logger.Info("Elasticsearch cluster status.", tag.ESIndex(index), tag.ESClusterStatus(status))
+	return nil
+}
+
+func (a *activities) UpdateClusterMetadataActivity(_ context.Context, params WorkflowParams) error {
+	oldSearchAttributes, err := a.saManager.GetSearchAttributes(params.IndexName, true)
+	if err != nil {
+		return fmt.Errorf("%w: %v", ErrUnableToGetSearchAttributes, err)
+	}
+
+	oldCustomSearchAttributes := searchattribute.FilterCustom(oldSearchAttributes)
+	newCustomSearchAttributes := map[string]enumspb.IndexedValueType{}
+	for saName, saType := range oldCustomSearchAttributes {
+		newCustomSearchAttributes[saName] = saType
+	}
+	for saName, saType := range params.CustomAttributesToAdd {
+		newCustomSearchAttributes[saName] = saType
+	}
+	err = a.saManager.SaveSearchAttributes(params.IndexName, newCustomSearchAttributes)
+	if err != nil {
+		a.logger.Info("Unable to save search attributes to cluster metadata.", tag.ESIndex(params.IndexName), tag.Error(err))
+		a.metricsClient.IncCounter(metrics.AddSearchAttributesWorkflowScope, metrics.AddSearchAttributesFailuresCount)
+		return fmt.Errorf("%w: %v", ErrUnableToSaveSearchAttributes, err)
```

Comment:
this err is retry-able?

Reply:
I guess it can be some transient DB error or version mismatch. Worth to retry in all cases (all standart go errors are retryable by default).